lib/algolia/getRecords.js

```diff
@@ -0,0 +1,186 @@
+const fs = require('fs');
+const path = require('path');
+const url = require('url');
+const algoliasearch = require('algoliasearch/lite');
+
+async function updateIndex() {
+    const appId = process.env.NEXT_PUBLIC_ALGOLIA_APP_ID;
+    const adminKey = process.env.NEXT_PUBLIC_ALGOLIA_ADMIN_API_KEY;
+    const algoliaIndex = process.env.NEXT_PUBLIC_ALGOLIA_INDEX;
+    const projectENV = process.env.NEXT_PUBLIC_APP_ENV;
+    if (appId && adminKey && algoliaIndex && projectENV === 'prod') {
+        const jsonDirectory = path.join(process.cwd());
+        const contentPath = `${jsonDirectory}/common`;
+        const docsPath = `${jsonDirectory}/docs`;
+        const basePath = `${url.pathToFileURL(jsonDirectory).toString()}/docs`;
+        const records = [];
+        const contentAlias = {};
+        const specialChars = [
+            '#',
+            '|',
+            '`',
+            '\\',
+            '--',
+            '\n',
+            '  ',
+            '[',
+            ']',
+            '!',
+            '{',
+            '}',
+            '- ',
+            '*',
+            '> Note :',
+            '<Tab',
+            '/>'
+        ];
+        const getPlatform = (pathString) => {

```

Comment:
why are we defining functions inside function ?

Reply:
Refactored the code